bigquery/integration_test.go

```diff
@@ -272,6 +275,44 @@ func initTestState(client *Client, t time.Time) func() {
 	}
 }
 
+// delete a resource if it is older than a day
+// that will prevent collisions with parallel CI test runs.
+func isResourceStale(t time.Time) bool {
+	return time.Now().Sub(t).Hours() >= 24
+}
+
+// delete old datasets
+func deleteDatasets(ctx context.Context, prefix string) {
+	it := client.Datasets(ctx)
+
+	for {
+		ds, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			fmt.Println("failed to list project datasets: ", err)
+			break
+		}
+		if !strings.HasPrefix(ds.DatasetID, prefix) {
+			continue
+		}
+
+		md, err := ds.Metadata(ctx)
+		if err != nil {
+			fmt.Printf("failed to get dataset `%s` metadata: %v", ds.DatasetID, err)
+			continue
+		}
+		if isResourceStale(md.CreationTime) {
```

Comment:
Why not [`time.Since()`](https://pkg.go.dev/time#Since) and avoid the extra function definition?

Reply:
I added the extra function so the logic for what is a stale resource can be reused. For example, later maybe we need to clean other resources like Storage Buckets. I changed to use `time.Since`